SYCL/ESIMD/esimd_test_utils.hpp

```diff
@@ -98,27 +99,86 @@ bool write_binary_file(const char *fname, const std::vector<T> &vec,
   return !ofs.bad();
 }
 
-template <typename T>
-bool cmp_binary_files(const char *fname1, const char *fname2, T tolerance) {
-  const auto vec1 = read_binary_file<T>(fname1);
-  const auto vec2 = read_binary_file<T>(fname2);
-  if (vec1.size() != vec2.size()) {
-    std::cerr << fname1 << " size is " << vec1.size();
-    std::cerr << " whereas " << fname2 << " size is " << vec2.size()
+template <typename T,
+          typename std::enable_if<std::is_integral<T>::value ||
+                                      std::is_floating_point<T>::value,
+                                  int>::type = 0>
+bool cmp_binary_files(const char *testOutFile, const char *referenceFile,
+                      const T tolerance = 0,
+                      const double mismatchRateTolerance = 0,
+                      const int mismatchReportLimit = 9) {
```

Comment:
@kbobrovs , maybe it's better to make it overridable via environment for convenience.

Reply:
Sounds good. Could be a separate patch